src/mono/wasm/debugger/BrowserDebugProxy/DevToolsProxy.cs

```diff
@@ -235,6 +238,9 @@ public async Task RunForDevTools(Uri browserUri, WebSocket ideSocket, Cancellati
 
                 ClientWebSocket browserSocket = new();
                 browserSocket.Options.KeepAliveInterval = Timeout.InfiniteTimeSpan;
+                var proxy = WebRequest.DefaultWebProxy;
+                if (_options is not null && _options.IgnoreProxyForLocalAddress && proxy is not null && !proxy.IsBypassed(browserUri))
```

Comment:
I'm not sure I understand. How is `IgnoreProxyForLocalAddress` being honored here, when we are not checking if the `browserUri` is a local address or not?

Reply:
Looking at https://github.com/dotnet/aspnetcore/pull/49395/files

`var ignoreProxyForLocalAddress = (noProxyEnvVar is not null && (noProxyEnvVar.Equals("localhost") || noProxyEnvVar.Equals("127.0.0.1"))) ? "--IgnoreProxyForLocalAddress True" : "";`

.. it is passing `true` as the argument if the url is local. So, the check is on blazor side and not the proxy here. Would it make more sense to name this `IgnoreProxy` then?

Reply:
I think browserUri is always a local address right?

Reply:
The app/browser could be running on a remote machine, IIUC. For example, in case of running the proxy on codespaces but the browser/app on a local machine.